ansible/roles/integration_tests/tasks/test_data.yml

```
@@ -2,33 +2,96 @@
+    - prepare-bundle
+  ansible.builtin.template:
+    src: ci.yaml.j2
+    dest: "{{ git_temp_dir.path }}/ci.yaml"
+    mode: 0644
+  changed_when: true
+
+- name: Prepare ci.yaml in target branch
   tags:
     - prepare-bundle
   ansible.builtin.shell: |
-    BASE_BRANCH="{{ git_base_branch }}"
-    UPSTREAM_BRANCH="{{ git_upstream_branch }}"
-    OPERATOR_PACKAGE_NAME="{{ operator_package_name }}"
-    SRC_BRANCH="{{ src_operator_git_branch }}"
-    BUNDLE_BRANCH="{{ git_bundle_branch }}"
-    SRC_VERSION="{{ src_operator_bundle_version }}"
-    DEST_VERSION="{{ operator_bundle_version }}"
+    UPSTREAM_BRANCH="{{ integration_tests_git_upstream_branch }}"
+    OPERATOR_PACKAGE_NAME="{{ integration_tests_operator_package_name }}"
+    SRC_BRANCH="{{ integration_tests_src_operator_git_branch }}"
+    BASE_BRANCH="{{ integration_tests_git_base_branch }}"
 
     git config user.name 'rh-operator-bundle-test-e2e'
     git config user.email 'exd-guild-isv+operators-test-e2e@redhat.com'
 
+    # Create a new target branch from the base branch
     git checkout -b $UPSTREAM_BRANCH $BASE_BRANCH
-    git push origin $UPSTREAM_BRANCH
 
-    git checkout -b $BUNDLE_BRANCH $BASE_BRANCH
-    git checkout origin/$SRC_BRANCH -- operators/$OPERATOR_PACKAGE_NAME
+    # Copy the ci.yaml from rendered template and push it to the target branch
+    mkdir -p "operators/$OPERATOR_PACKAGE_NAME"
+    cp ci.yaml "operators/$OPERATOR_PACKAGE_NAME/ci.yaml"
+
+    git add "operators/$OPERATOR_PACKAGE_NAME/ci.yaml"
+    git commit -m "Copied ci.yaml from $SRC_BRANCH"
+    git push origin "$UPSTREAM_BRANCH"
+  args:
+    executable: /bin/bash
+    chdir: "{{ git_temp_dir.path }}"
+  changed_when: true
+
+- name: Generate and push Operator Bundle
+  tags:
+    - prepare-bundle
+  ansible.builtin.shell: |
+    UPSTREAM_BRANCH="{{ integration_tests_git_upstream_branch }}"
+    SRC_OPERATOR_PACKAGE_NAME="{{ integration_tests_src_operator_package_name }}"
+    OPERATOR_PACKAGE_NAME="{{ integration_tests_operator_package_name }}"
+
+    # A branch from where the bundle will be copied
+    SRC_OPERATOR_BRANCH="{{ integration_tests_src_operator_git_branch }}"
+
+    # A branch where the bundle will be pushed
+    BUNDLE_BRANCH="{{ integration_tests_git_bundle_branch }}"
+
+    # A version of the bundle to be copied from the source branch
+    SRC_VERSION="{{ integration_tests_src_operator_bundle_version }}"
+
+    # A version of the bundle to be pushed to the target branch
+    DEST_VERSION="{{ integration_tests_operator_bundle_version }}"
+
+    # A flag to determine if the bundle is for FBC catalog submission
+    FBC_CATALOG="{{ integration_tests_fbc_catalog }}"
+
+    echo "FBC catalog is: $FBC_CATALOG"
+
+    if [[ $FBC_CATALOG == "True" ]]; then
+      # In case of FBC catalog submission a bundle needs to be pushed in the target branch
+      BUNDLE_BRANCH=$UPSTREAM_BRANCH
+      git checkout $BUNDLE_BRANCH
+    else
+      git checkout -b $BUNDLE_BRANCH $UPSTREAM_BRANCH
```

This command could break if there were some mistakes in the `BUNDLE_BRANCH` or `UPSTREAM_BRANCH` variables. If they contained a space for any reason, this would be interpreted as multiple words/arguments instead of one. Best practice is to double-quote shell variable expansion.  This same problem occurs at more places in this file.
```suggestion
      git checkout -b "$BUNDLE_BRANCH" "$UPSTREAM_BRANCH"
```

---

Fixed